### PR TITLE
http: add support for proxy

### LINKED ISSUE #3232: Add support for http/https proxy
## Description of the problem
In newer deployments, there is resistance to the use of HTTP destination using a firewall rule. Adoption would be easier if an explicit proxy was supported

## Proposed solution
Enhance the configuration of syslog-ng to pass proxy option to the libcurl component

### PR DESCRIPTION
**Pull request by vatsal-encs**
fixes #3232

### CODE REVIEW

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

**Comment by MrAnno** [comment]
@kira-syslogng ok to test

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by furiel** [comment]
Thank you for your pull request. At first glance it looks good to me.
- Please don't forget to free the proxy string.
- Please add a news file.
- I mentioned in the issue that it would be nice to have a warning if users use different proxy scheme than the url.

Unfortunately, I have a few things to do now, but i try to come back with a proper review as soon as possible. In the meantime, could you tell your experiences? It would help the review greatly.

- How did you test the feature? What proxy implementation did you use, and how did you configure it? 
- Can two different http destinations have different proxys? Is it possible to have one http destination with proxy, and another without proxy?
- The proxy setting in the grammar indeed stronger than the proxy environment variables?
- What happens if we have only one http destination, but have two urls and two workers? Do both workers use the same proxy?
- What happens to the messages if the proxy is not available? Does syslog-ng drop the messages?
- What happens to the messages if the proxy is available, but the remote isn't?

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: modules/http/http.h [hunk 1]
```diff
@@ -52,6 +52,7 @@ typedef struct
   gchar *cert_file;
   gchar *key_file;
   gchar *ciphers;
+  gchar *proxy;
```
[the remark this answers is not in the document]
Comment (vatsal-encs) [reply modules/http/http.h:55]: @furiel : okay

**Comment by furiel** [comment]
Thank you. Finally, please add a news file:
https://github.com/syslog-ng/syslog-ng/blob/master/CONTRIBUTING.md#news-file

**Comment by vatsal-encs** [comment]
> Thank you. Finally, please add a news file:
> https://github.com/syslog-ng/syslog-ng/blob/master/CONTRIBUTING.md#news-file

@furiel : Okay, I have added it.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by furiel** [comment]
Thank you. The patchset looks good now.

Unfortunately pytest-linters is broken, and as I see it is a bug in virtualenv, that the maintainers are currently trying to address in https://github.com/pypa/virtualenv/pull/1811. We can wait a little until it is resolved. Or we can just merge this as it is: python code has not been changed, so the linters should not find any problem.

**Comment by alltilla** [comment]
Merging, as travis fails for an environment error.

**Pull request merged**